x-pack/plugins/synthetics/public/apps/synthetics/state/monitor_details/index.ts

```diff
@@ -42,7 +48,17 @@ export const monitorDetailsReducer = createReducer(initialState, (builder) => {
     .addCase(setMonitorDetailsLocationAction, (state, action) => {
       state.selectedLocationId = action.payload;
     })
-
+    .addCase(getMonitorLastRunAction.get, (state, action) => {
+      state.lastRun.loading = true;
+      if (checkIsStalePing(action.payload.monitorId, state.lastRun.data)) {
+        state.lastRun.data = undefined;
+      }
+    })
+    .addCase(getMonitorLastRunAction.success, (state, action) => {
+      state.lastRun.loading = false;
+      state.lastRun.data = action.payload.pings[0];
```

Comment:
```suggestion
      state.lastRun.data = action.payload.pings?.[0];
```

Reply:
The type checker should have caught that :/ Let me revise what the response is like when there are no results and see if we need to update the types.

Reply:
I see the types are correct. The `pings` array will just be empty so the null check won't do anything here.

<img width="372" alt="Screenshot 2022-11-01 at 11 01 23" src="https://user-images.githubusercontent.com/57448/199208793-d460a010-243b-41a8-9a22-7cdcabd48f7c.png">